### DISCUSSION TITLE
How to use cmdPosition to control 20 crazyflies to take off?

### DISCUSSION THREAD

**Discussion by vegetable-to-kou-foot** [USC-ACTLab/crazyswarm#359] [Q&A]
Hi, @whoenig @jpreiss
I'm trying to use cmdPosition to control 20 crazyflies to take off, but I don't know where the problem is. Here are three tests.

# Test 1

I'm trying to control 20 crazyflies with one crazyradio PA. Sometimes it takes off successfully, but often it fails. Here is an example of failure:

1. The number of crazyflies is: No.1-10 for the first line from left to right, and No.11-20 for the second line from left to right.

2. There should be no other 2.4GHz interference nearby, such as WiFi or Bluetooth.

3. All the crazyflies are in ch30.

4. I'm using the latest firmware.

5. Nicehover.py works very well. It can control 20 crazyflies to take off and land at the same time.

6. Our program takes off and lands using cmdPosition. We discretize the flight path into some points with an interval of 0.016m, and send the next point to each CF every 0.1s, so as to realize take-off and landing. It is very similar to example_ cmd_ pos.py.

7. We use timeHelper.sleep(8.0) to sleep for 8s after creating swarm.allcfs, otherwise all the planes may lose control.

8. During take-off, there are often 2-6 crazyflies rollover, and it is usually some of the No.7-14 in a row. As shown in the video, this time, the No.7 and No.8 are out of control (ignore the No.15, it has no power).It seems that the communication failure of No.7 caused the communication failure of No.8 or even more.

# Test 2

Then, we tried to control 20 crazyflies with two crazyradio PA. We change the channel of No.1-10 to ch20, and No.11-20 are still in ch30. Two crazyradio PA are plugged into the adjacent USB interface, as shown in figure.This time, some crazyflies still couldn't take off. Worse than test 1, even if they took off successfully, the movements of these crazyflies did not seem to be continuous.

# Test 3

On the basis of test 2, we moved one of the two radio stations to 1m away with USB docking station. This time, we have successfully taken off and landed for two times, and there is no problem that the movement seems to be discontinuous. But when we raised the radio, the problem in test 2 came back.

# Question

How to repair the communication failure of an crazyflie, which leads to the communication failure of several crazyflies behind it? For example, it may be possible to resend data packets and try to restore communication.Or each crazyflie may not be affected by the communication status of the previous one.

# Video and figure

https://user-images.githubusercontent.com/34856871/113906512-eab6b400-9806-11eb-99fe-e745382c8511.mp4

![figure1](https://user-images.githubusercontent.com/34856871/113906544-f1452b80-9806-11eb-8f5d-107cd43bf030.jpg)

**Comment by swarm5** [comment]
Hello @whoenig , @vegetable-to-kou-foot is my colleague. As we communicated before, we have encountered big trouble with the use of `cmdposition() `to control the takeoff of 20 cfs.
We use `timeHelper.sleep(8.0)` to sleep for 8s after creating swarm.allcfs, this operation works well. We have about 60% probability that we can use `cmdposition()` to take off 20 cfs. However, sometimes **a few consecutive cfs randomly appeared** and did not respond to take-off instructions.


Recently, he and I are hoping to display a swarm of 20 cfs on the day of the laboratory’s open day, so we can’t wait for the new communication mechanism of crazyswarm. We hope to fix the existing problems.

【problem】:
The characteristics of our current problem are very obvious: **often several cfs with consecutive ID do not respond to the `cmdposition` take-off command,** we would like to ask, what is the likely cause of this problem? Or which programs may have errors, we can narrow the scope to solve existing problems. (Perhaps not just the `cmdposition` takeoff command, but the low-level command that just started. Because we used the high-level command `takeoff` to take off before, we continue to use `cmdposition` to control, there will be some cfs hanging in the air.)

**Reply by whoenig** [reply]
This is very odd. Could you make sure that you have the latest NRF firmware. In particular one, that has the following bugfix: https://github.com/bitcraze/crazyflie2-nrf-firmware/commit/4bfc7ea18ab4452ff245fff58ae714e24f06336f?
